src/amo/components/Categories.js

```diff
@@ -1,29 +1,53 @@
 import React, { PropTypes } from 'react';
+import { connect } from 'react-redux';
 import { compose } from 'redux';
+import { showLoading } from 'react-redux-loading-bar';
 
 import Link from 'amo/components/Link';
-import { visibleAddonType } from 'core/utils';
+import { categoriesFetch } from 'core/actions/categories';
+import { apiAddonType, visibleAddonType } from 'core/utils';
 import translate from 'core/i18n/translate';
+import LoadingText from 'ui/components/LoadingText';
 
 import './Categories.scss';
 
 
 export class CategoriesBase extends React.Component {
   static propTypes = {
     addonType: PropTypes.string.isRequired,
+    dispatch: PropTypes.func,
     categories: PropTypes.object.isRequired,
+    clientApp: PropTypes.string.isRequired,
     error: PropTypes.bool,
     loading: PropTypes.bool.isRequired,
     i18n: PropTypes.object.isRequired,
   }
 
+  componentWillMount() {
+    const { addonType, categories, clientApp, dispatch } = this.props;
+    if (!Object.values(categories).length) {
```

Comment:
I would say this is the big difference for redux-saga: our components are now expected to load/work without data and then dispatch their own actions to get said data if that's the case. For this simple case of categories it was easy and I ended up reloading them from state because they nearly never change and it made sense. For things like search I suspect there'd be more effort involved though in reality it will mostly be shuffling around existing code. But I think this component now makes a lot more sense to look at: if no categories exist (basically only happens on first load) it fetches them and shows a placeholder until then. The mechanism for loading is in the saga, yes, but I think this is a bit easier to follow.